optuna/study/_dataframe.py

```diff
@@ -53,15 +56,21 @@ def _create_records_and_aggregate_column(
                     column_agg[attr].add((df_column, nested_attr))
             elif isinstance(value, list):
                 # Expand trial.values.
-                for nested_attr, nested_value in enumerate(value):
+                iterator = enumerate(value) if metric_names is None else zip(metric_names, value)
+                for nested_attr, nested_value in iterator:
                     record[(df_column, nested_attr)] = nested_value
                     column_agg[attr].add((df_column, nested_attr))
             elif attr == "values":
                 # trial.values should be None when the trial's state is FAIL or PRUNED.
                 assert value is None
-                for nested_attr in range(len(study.directions)):
+                iterator = range(len(study.directions)) if metric_names is None else metric_names
+                for nested_attr in iterator:
                     record[(df_column, nested_attr)] = None
                     column_agg[attr].add((df_column, nested_attr))
+            elif attr == "value":
+                nested_attr = non_nested_attr if metric_names is None else metric_names[0]
+                record[(df_column, nested_attr)] = value
+                column_agg[attr].add((df_column, nested_attr))
```

Comment:
Could you explain what these codes intend to do?

Reply:
This block describes the processing when the given attr is specifically the value of a trial. Since the value is of type float, it cannot be merged with the processing of attr of type list as is. Therefore, it is written in this way.

Reply:
I couldn't be sure that the value of `value` is necessarily either `trial.value` or `trial.values` in https://github.com/optuna/optuna/pull/4383/files/ecd8c254ff5701a9e681c30c6bde307600296a10#diff-39d0567a9ce90fcd4881e6f45e621c3cd0126c7864e7c54a0772400601ad3387R57. 
If not, it looks like you are doing the transformation of https://github.com/optuna/optuna/pull/4383/files/ecd8c254ff5701a9e681c30c6bde307600296a10#diff-39d0567a9ce90fcd4881e6f45e621c3cd0126c7864e7c54a0772400601ad3387R59 even when `value` is not `trial.values` but its type happens to be a `list`, which looks wrong.
Am I misunderstanding something?